js/clinicalTimeline.js

```diff
@@ -163,16 +165,46 @@ var clinicalTimeline = (function(){
       .attr("width", overviewAxisWidth)
       .attr("class", "overview");
 
-    clinicalTimelineOverviewAxis(overviewSVG, getTickValues, minDays, maxDays, overviewAxisWidth, formatTime, daysToTimeObject, getZoomLevel, width,  margin);
-    clinicalTimelineVerticalLine(timelineBool.enableVerticalLine, beginning, ending, timelineBool.advancedView, timelineBool.tooltipOnVerticalLine);
-    initHelperLines(timelineBool.enableHelperLines, zoomFactor, timelineBool.advancedView, divId, formatTime, daysToTimeObject);
+    //object to be shared by all plugins
+    var options = {
+      timelineBool: timelineBool,
+      divId: divId,
+      beginning: beginning,
+      ending: ending,
+      minDays: minDays,
+      maxDays: maxDays,
+      width: width,
+      margin: margin,
+      getZoomLevel: getZoomLevel,
+      getZoomFactor: getZoomFactor,
+      zoomFactor: zoomFactor,
+      getTickValues: getTickValues,
+      formatTime: formatTime,
+      daysToTimeObject: daysToTimeObject,
+      chart: chart,
+      overviewAxisWidth: overviewAxisWidth,
+      overviewX: overviewX,
+      startAllowedOverview: startAllowedOverview,
+      endAllowedOverview: endAllowedOverview,
+      timeline: timeline,
+      clinicalTimelinePlugins: clinicalTimelinePlugins
+    }
```

Comment:
I'm still wondering how much you need this `options` object. It seems like we are now exposing one clinicalTimeline API to the user through the `timeline` object and another more extensive API with the `options` object to each plugin. I think it would make sense to consolidate that to a single API on the timeline object. It's not super clear to me why we need two different ones.